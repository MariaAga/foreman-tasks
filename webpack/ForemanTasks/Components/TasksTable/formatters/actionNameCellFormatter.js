import React from 'react';
import { cellFormatter } from 'foremanReact/components/common/table';
import EllipsisWithTooltip from 'react-ellipsis-with-tooltip';

export const actionNameCellFormatter = url => (value, { rowData: { id } }) =>
  cellFormatter(
    <a href={`/${url}/${id}`}>
      <EllipsisWithTooltip>{value}</EllipsisWithTooltip>
    </a>
  );
