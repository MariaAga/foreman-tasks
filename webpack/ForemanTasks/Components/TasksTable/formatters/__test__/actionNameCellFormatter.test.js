import { actionNameCellFormatter } from '../actionNameCellFormatter';

describe('actionNameCellFormatter', () => {
  it('render', () => {
    const data = ['action-name', { rowData: { id: 'some-id' } }];
    expect(actionNameCellFormatter('some-url')(...data)).toMatchSnapshot();
  });
});
