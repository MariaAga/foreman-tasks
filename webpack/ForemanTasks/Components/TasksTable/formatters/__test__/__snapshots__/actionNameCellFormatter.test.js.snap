// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actionNameCellFormatter render 1`] = `
<a
  href="/some-url/some-id"
>
  action-name
</a>
`;
